WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m

```diff
@@ -84,10 +85,15 @@ - (XCElementSnapshot *)fb_lastSnapshot
   if (uniqueTypes && [uniqueTypes count] == 1) {
     type = [uniqueTypes.firstObject intValue];
   }
-  [matchedElements addObjectsFromArray:[[self descendantsMatchingType:type] matchingPredicate:[FBPredicate predicateWithFormat:@"%K IN %@", FBStringify(XCUIElement, wdUID), matchedUids]].allElementsBoundByIndex];
+  XCUIElementQuery *query = [[self descendantsMatchingType:type] matchingPredicate:[FBPredicate predicateWithFormat:@"%K IN %@", FBStringify(XCUIElement, wdUID), matchedUids]];
+  if (1 == snapshots.count) {
```

Comment:
I think this will resolve whole query anyway, so you have no benefit of using it here. Unless count is optimised here.